.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+# This GitHub workflow is triggered when a new tag is pushed.
+# A release is then created from this tag.
+
+name: Release
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get pushed tag
+        id: get_tag
+        run: echo ::set-output name=TAG::${GITHUB_REF/refs\/tags\//}
+
+      - name: Download repository archive at pushed tag
+        run: "curl -L https://github.com/${{ github.repository_owner }}/${{ github.event.repository.name }}/archive/refs/tags/${{ steps.get_tag.outputs.TAG }}.tar.gz -o rules_docker-${{ steps.get_tag.outputs.TAG }}.tar.gz"
+
+      - name: Get SHA256 of archive
+        id: get_sha
+        run: echo "::set-output name=sha::$(sha256sum rules_docker-${{ steps.get_tag.outputs.TAG }}.tar.gz | cut -f 1 -d ' ')"
+
+      - name: Get version
+        id: get_version
+        run: | 
+          VERSION=${{ steps.get_tag.outputs.TAG }}
+          echo "VERSION=${VERSION:1}" >> $GITHUB_ENV
+
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        with:
+          body: |
+            Copy the following into your WORKSPACE file to use rules_docker at release ${{ steps.get_tag.outputs.TAG }}
+            ```
+            http_archive(
+                name = "io_bazel_rules_docker",
+                sha256 = "${{ steps.get_sha.outputs.sha }}",
+                strip_prefix = "rules_docker-${{ env.VERSION }}",
```

Comment:
I think we may be able to remove the need for the strip_prefix?

Reply:
Hmm.

We could use [git archive](https://git-scm.com/docs/git-archive)?

```yaml
- uses: actions/checkout@v3
- run: git archive --output=rules_docker-${{ steps.get_tag.outputs.TAG }}.tar.gz ${{ steps.get_tag.outputs.TAG }}
```

Reply:
Yeah doing that in a follow up makes sense. I can probably look at it, but won't get a chance this week. Could be good to try what we have here so far with the next release anyway.

Reply:
Are you happy to try the git archive thing? It would be really nice to remove the requirement for `strip_prefix` as most other rulesets don't for major releases. If you don't have time, I'm happy to add a commit?

Reply:
I can probably get to this next week but if you'd like to try it in the meantime, feel free to add a commit. Otherwise, I'll try next week.

Reply:
@uhthomas actually had some time this evening & pushed the change so `strip_prefix` isn't needed. Your suggestion worked. Basically, the extracted parent dir just needs to match the name of the tar & then `strip_prefix isn't needed` so `git archive` works a treat. Thank you!